### PR TITLE
Remove deprecated code

### PR DESCRIPTION
**Pull request by RainerZufall187**

### CODE REVIEW

**Comment by PerchunPak** [comment]
Can you also remove other deprecated parts?
https://github.com/search?q=repo%3Apy-mine%2Fmcstatus%202023&type=code

**Review by ItsDrike — DISMISSED** [review]
Revert removing the deprecated decorator in `mcstatus/utils.py`, and the removed tests for it, we still want to keep these for future deprecations.

> why is there so much code for warning people about deprecated stuff

It's important for libraries to give people time to migrate their code-bases when something gets changed. It's much more pleasant for users to just see a warning like "Function X is deprecated, use function Y instead", then just getting a hard error saying that X doesn't exist, with no other info.

**Comment by kevinkjt2000** [comment]
> why is there so much code for warning people about deprecated stuff

AFAIK the Python standard library does not include a nice deprecation wrapper like the one we have.

**Comment by PerchunPak** [comment]
@RainerZufall187 can I take this PR?

File: pyproject.toml [hunk 1]
```diff
@@ -37,7 +37,6 @@ dnspython = "^2.4.2"
 [tool.poetry.group.dev.dependencies]
 pre-commit = "^3.3.3"
 tox = "^3.28.0"
-tox-poetry = "0.5.0" # deprecated, no sense in `^`
```
Comment (kevinkjt2000) [inline pyproject.toml:40]: Leave this in here, please. GitHub Actions still use it.

**Review by kevinkjt2000 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (PerchunPak) [reply pyproject.toml]: Returned in 34d6a70c4441660287e56c07f4a5bd546759b7e9

**Review by kevinkjt2000 — APPROVED** [review]

**Pull request merged**